TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/Drivebase.java

```diff
@@ -0,0 +1,275 @@
+     */
+    public @Api void addTelemetry(Telemetry telemetry) {
+        telemetry.addData("FLDrive", fldrive.getCurrentPosition());
+        telemetry.addData("FRDrive", frdrive.getCurrentPosition());
+        telemetry.addData("BLDrive", bldrive.getCurrentPosition());
+        telemetry.addData("BRDrive", brdrive.getCurrentPosition());
+        telemetry.addData("Heading", getHeading());
+        telemetry.update();
+    }
+
+    /**
+     * For use wherever.
+     * This resets the drivebase encoders.
+     * Called before `waitForStart()` and after drivebase init.
+     *
+     * @param choice If passed, what mode to set the motors to after resetting them, otherwise whatever they were before.
+     */
+
+    public @Api void clearEncoder(DcMotor.RunMode choice) {
+        DcMotor.RunMode oldMode = choice == null ? fldrive.getMode() : choice;
+
+        setMotorModes(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        setMotorModes(oldMode);
+    }
+
+
+    /**
+     * For use in teleop, controlled by a controller.
+     *
+     * @param yInput    Driving input. Negative is back, positive is forward. [-1, 1].
+     * @param xInput    Strafing input. Negative is left, positive is right. [-1, 1].
+     * @param turnInput Turning input. Negative is ccw, positive is clockwise. [-1, 1].
+     */
+    public @Api void mecanumDrive(double yInput, double xInput, double turnInput) {
+        double frontLeft = yInput + xInput + turnInput;
+        double frontRight = yInput - xInput - turnInput;
+        double backLeft = yInput - xInput + turnInput;
+        double backRight = yInput + xInput - turnInput;
+
+        double clamp = maxOf(1, frontLeft, frontRight, backLeft, backRight);
+
+        fldrive.setPower(frontLeft / clamp);
+        frdrive.setPower(frontRight / clamp);
+        bldrive.setPower(backLeft / clamp);
+        brdrive.setPower(backRight / clamp);
+    }
+
+
+    /**
+     * For use in auto. Drives forward a specified distance, using the encoders.
+     *
+     * @param inches    The amount of inches to drive forward, negative is backwards.
+     * @param power     How fast to drive. (0, 1].
+     * @param telemetry Pass telemetry if you want this method to log to telemetry.
+     * @see Drivebase#driveSideways
+     * @see Drivebase#turnAngle
+     * @see Drivebase#turnToAngle
+     */
+    public @Api void driveForward(double inches, double power, @Nullable Telemetry telemetry) {
+        setMotorModes(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+
+        double target = inches * ENCODER_PER_INCH;
+
+        fldrive.setTargetPosition((int) target);
+        frdrive.setTargetPosition((int) target);
+        bldrive.setTargetPosition((int) target);
+        brdrive.setTargetPosition((int) target);
+
+        fldrive.setPower(power);
+        frdrive.setPower(power);
+        bldrive.setPower(power);
+        brdrive.setPower(power);
+
+        setMotorModes(DcMotor.RunMode.RUN_TO_POSITION);
```

Comment:
When setting DcMotor.RunMode.RUN_TO_POSITION, you need to set the motor powers afterwards. 
Below is an example of this, taken from reference code, with the moveRobot function, being what sets the motor powers to the maxDriveSpeed variable.
```java
            // Set Target FIRST, then turn on RUN_TO_POSITION
            flDrive.setTargetPosition(flTarget);
            frDrive.setTargetPosition(frTarget);
            blDrive.setTargetPosition(blTarget);
            brDrive.setTargetPosition(brTarget);

            flDrive.setMode(DcMotor.RunMode.RUN_TO_POSITION);
            frDrive.setMode(DcMotor.RunMode.RUN_TO_POSITION);
            blDrive.setMode(DcMotor.RunMode.RUN_TO_POSITION);
            brDrive.setMode(DcMotor.RunMode.RUN_TO_POSITION);

            // Set the required driving speed  (must be positive for RUN_TO_POSITION)
            // Start driving straight, and then enter the control loop
            maxDriveSpeed = Math.abs(maxDriveSpeed);
            moveRobot(maxDriveSpeed, 0);
```

Reply:
fixed